xcube/cli/gen.py

```diff
@@ -110,41 +116,40 @@ def _handle_error(e, traceback_mode):
+# https://stackoverflow.com/questions/47437472/in-python-click-how-do-i-see-help-for-subcommands-whose-parents-have-required
 
-class GenL2CHelpFormatter(argparse.HelpFormatter):
-
-    def format_help(self):
-        # noinspection PyUnresolvedReferences
-        help_text = super().format_help()
-
-        input_processors = get_obj_registry().get_all(type=InputProcessor)
-        output_writers = query_dataset_io(lambda ds_io: 'w' in ds_io.modes)
-
-        help_text += '\ninput processors to be used with option --proc:\n'
-        help_text += self._format_input_processors(input_processors)
-        help_text += '\noutput formats to be used with option --writer:\n'
-        help_text += self._format_dataset_ios(output_writers)
-        help_text += '\n'
-
-        return help_text
-
-    @classmethod
-    def _format_input_processors(cls, input_processors):
-        help_text = ''
-        for input_processor in input_processors:
-            fill = ' ' * (34 - len(input_processor.name))
-            help_text += f'  {input_processor.name}{fill}{input_processor.description}\n'
-        return help_text
-
-    @classmethod
-    def _format_dataset_ios(cls, dataset_ios):
-        help_text = ''
-        for ds_io in dataset_ios:
-            fill1 = ' ' * (24 - len(ds_io.name))
-            fill2 = ' ' * (10 - len(ds_io.ext))
-            help_text += f'  {ds_io.name}{fill1}(*.{ds_io.ext}){fill2}{ds_io.description}\n'
-        return help_text
-
-
-if __name__ == '__main__':
-    sys.exit(main())
+# class GenL2CHelpFormatter(argparse.HelpFormatter):
+#
+#     def format_help(self):
+#         # noinspection PyUnresolvedReferences
+#         help_text = super().format_help()
+#
+#         input_processors = get_obj_registry().get_all(type=InputProcessor)
+#         output_writers = query_dataset_io(lambda ds_io: 'w' in ds_io.modes)
+#
+#         help_text += '\ninput processors to be used with option --proc:\n'
+#         help_text += self._format_input_processors(input_processors)
+#         help_text += '\noutput formats to be used with option --writer:\n'
+#         help_text += self._format_dataset_ios(output_writers)
+#         help_text += '\n'
+#
+#         return help_text
+#
+#     @classmethod
+#     def _format_input_processors(cls, input_processors):
+#         help_text = ''
+#         for input_processor in input_processors:
+#             fill = ' ' * (34 - len(input_processor.name))
+#             help_text += f'  {input_processor.name}{fill}{input_processor.description}\n'
+#         return help_text
+#
+#     @classmethod
+#     def _format_dataset_ios(cls, dataset_ios):
+#         help_text = ''
+#         for ds_io in dataset_ios:
+#             fill1 = ' ' * (24 - len(ds_io.name))
+#             fill2 = ' ' * (10 - len(ds_io.ext))
+#             help_text += f'  {ds_io.name}{fill1}(*.{ds_io.ext}){fill2}{ds_io.description}\n'
+#         return help_text
```

Comment:
Please find a solution how to display this info. If there is absolutley no chance to customize click, then we need to introduce a new command, such as `info` with special options that will print out the desired information, e.g.

    $ xcube info --formats
    $ xcube info --iprocs

Reply:
This will be solved in a different branch as mentioned in the pull request text at the very top. It is also remarked in with a TODO in this branch.

Reply:
I do not agree. This feature is mandatory and cannot be dropped in our code on the master branch.